### DISCUSSION TITLE
Focus on newly spawned window

### DISCUSSION THREAD

**Discussion by KulkarniKaustubh** [hyprwm/Hyprland#1793] [Q&A]
So I didn't know whether to create an issue or not. Basically what happens is that, when I open a new window, say, a terminal, for example, I want the keyboard focus on the newly spawned window no matter where my mouse is. On the dwindle layout, I have forced a right side split. So if my mouse is on the left side of the workspace, any new spawned windows don't have keyboard focus.

1. Is this intended behavior?
2. If it is, is there any way to change it through the config file?

Thanks!

**Comment by vaxerski** [comment]
The focus will go to the newly spawned window. If you have follow_mouse turned on though, any movement of the mouse will refocus the window under your mouse.

**Reply by KulkarniKaustubh** [reply]
I do have `follow_mouse = 1`, but the newly spawned window doesn't gain focus even without me touching the mouse.

One thing to note here, I just experimented a little and found something consistent. If I have emacs open and spawn anything else, the focus remains on emacs.

**Reply by vaxerski** [reply]
and what if you have anything other than emacs open?

**Reply by KulkarniKaustubh** [reply]
With anything other than emacs, it works perfectly fine. Looks like something specific to emacs windows. Is this fixable through any changes in Hyprland or shall I close this discussion?

Thanks for the prompt replies!

**Reply by vaxerski** [reply]
do you have focus on activate enabled? maybe that's the issue?

**Reply by KulkarniKaustubh** [reply]
Nope, I have not explicitly set focus on active to enabled anywhere in my config file.

**Reply by CalfMoon** [reply]
I'm having the same problem as the author only with emacs. I have explicitly set focus on activate to enabled and nothing changed.
I recorded my screen and check the focus transfers to new window just for a split second and then goes back to emacs.
If the author or anyone has found the solution please post it.

**Reply by linuxer0x42** [reply]
Same issue with me, only happens when Emacs is open though, if I switch to another workspace the focus goes back to the Master window like it should.

**Reply by DaeTheDerg** [reply]
I also seem to be experiencing this issue, with all windows though

**Reply by DaeTheDerg** [reply]
~oops, forgot to add `focus_on_activate = true` to my config~

Edit: nevermind, it's still not working

**Reply by aloispichler** [reply]
same issue here, with Emacs only.

**Reply by night-kat** [reply]
I have a similar problem. I want clipse to autofocus when launching it but the focus always stays where it was, even with ```focus_on_activate = true``` and ```follow_mouse = 2```

**Reply by 0WD0** [reply]
```
misc {
    focus_on_activate = true
}
```

solved my problem on emacs.

**Comment by azizsayfiddin** [comment]
> So I didn't know whether to create an issue or not. Basically what happens is that, when I open a new window, say, a terminal, for example, I want the keyboard focus on the newly spawned window no matter where my mouse is. On the dwindle layout, I have forced a right side split. So if my mouse is on the left side of the workspace, any new spawned windows don't have keyboard focus.
> 
>     1. Is this intended behavior?
> 
>     2. If it is, is there any way to change it through the config file?
> 
> 
> Thanks!

Maybe the `follow_mouse = 2` in the input{...} can help?

**Reply by linuxer0x42** [reply]
This does not fix the issue for me, with Emacs open, it still focuses the window under the mouse when I open a new window.

**Comment by rupertpaulson** [comment]
This worked for me:
https://wiki.hypr.land/Configuring/Dwindle-Layout/
```
dwindle {
    force_split = 2
}

# 0 > split follows mouse (DEFAULT CAUSING THE ISSUE)
# 1 > always split to the left (new = left or top)
# 2 > always split to the right (new = right or bottom)

```
